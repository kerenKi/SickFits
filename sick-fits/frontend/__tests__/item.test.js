import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';

const fakeItem = {
    id: 'ABC123',
    title: 'test item',
    description: 'An item for testing',
    image: 'image.jpg',
    price: 5000,
    largeImage: 'largeImage.jpg',
}

describe('<Item/>', () => {

    it('renders the priceTag and title properly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem}/>)
        // console.log(wrapper.debug());
        const PriceTag = wrapper.find('PriceTag')
        // console.log('PriceTag.text()', PriceTag.text());
        // console.log('PriceTag.dive().text()', PriceTag.dive().text());
        // console.log({
        //     children: PriceTag.children().debug()
        // });
        expect(PriceTag.children().text()).toBe('$50')
        expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
    
    })

    it('renders the image properly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem}/>)
        const image = wrapper.find('img')
        // console.log(image.props());
        expect(image.props().alt).toBe(fakeItem.title)
    })

    it('renders the buttons properly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem}/>)
        const buttonList = wrapper.find('.buttonList')
        console.log(buttonList.debug());
        console.log(buttonList.children().debug());
        //check that buttonList has 3 buttons
        expect(buttonList.children()).toHaveLength(3)
        //check that Link is one of the buttons on buttonList
        expect(buttonList.find('Link')).toHaveLength(1)

        //check that AddToCart is one of the buttons on buttonList
        expect(buttonList.find('AddToCart').exists()).toBe(true)

        //check that DeleteItem is one of the buttons on buttonList
        expect(buttonList.find('DeleteItem').exists()).toBe(true)
    })
})