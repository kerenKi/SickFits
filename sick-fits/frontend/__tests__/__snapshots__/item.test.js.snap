// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="test item"
    src="image.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        test item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    An item for testing
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
